gnd/src/main/java/com/google/android/gnd/ui/editobservation/AbstractFieldViewModel.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.editobservation;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.LiveDataReactiveStreams;
+import androidx.lifecycle.MutableLiveData;
+import com.google.android.gnd.model.form.Field;
+import com.google.android.gnd.model.observation.Response;
+import com.google.android.gnd.ui.common.AbstractViewModel;
+import io.reactivex.Single;
+import io.reactivex.processors.BehaviorProcessor;
+import java8.util.Optional;
+
+public class AbstractFieldViewModel extends AbstractViewModel {
+
+  private final LiveData<Optional<Response>> response;
+  private final LiveData<String> responseText;
+  private final MutableLiveData<String> error = new MutableLiveData<>();
+  private final BehaviorProcessor<Optional<Response>> responseSubject = BehaviorProcessor.create();
+
+  private Field field;
+
+  AbstractFieldViewModel() {
```

Comment:
Please add Javadoc to classes,, public methods, and especially to streams like the ones below to clarify what they're used for and why.